File: src/domain/entities/Estudiante.php
<?php

namespace src\domain\entities;

use DateTime;

// Se crea la clase 'Estudiante'.
class Estudiante {
    private int $id;
    private string $nombre;
    private string $apellido;
    private DateTime $fechaNacimiento;
    private int $semestre;
    private string $email;
    private string $genero;
    private string $telefono;
    private string $programa;
    private string $universidad;


    // -- Constructores. --
    public function __construct(
        int $id,
        string $nombre,
        string $apellido,
        DateTime $fechaNacimiento,
        int $semestre,
        string $email,
        string $genero,
        string $telefono,
        string $programa,
        string $universidad
    
    )


    {
        $this->id = $id;
        $this->nombre = $nombre;
        $this->apellido = $apellido;
        $this->fechaNacimiento = $fechaNacimiento;
        $this->semestre = $semestre;
        $this->email = $email;
        $this->genero = $genero;
        $this->telefono = $telefono;
        $this->programa = $programa;
        $this->universidad = $universidad;
        
        // Asegura cumplir las reglas minimas.         
        $this->validar();

    }


    // Reglas basicas de validación del 'dominio'.
    private function validar(): void{
        if ($this->semestre < 1) {
            throw new \InvalidArgumentException("El semestre debe ser mayor o igual a 1.");
        }

        if (!filter_var($this->email, FILTER_VALIDATE_EMAIL)){
            throw new \InvalidArgumentException("El correo electronico no es válido.");
        }

        if (strlen($this->telefono) < 7) {
            throw new \InvalidArgumentException("El teléfono debe tener al menos 7 digitos.");

        }

    }


    # -- Getters. --
    public function getId(): int{
        return $this->id;
    }

    public function getNombre(): string{
        return $this->nombre;
    }

    public function getApellido(): string{
        return $this->apellido;
    }

    public function getFechaNacimiento(): DateTime{
        return $this->fechaNacimiento;
    }

    public function getSemestre(): int{
        return $this->semestre;
    }

    public function getEmail(): string{
        return $this->email;
    }

    public function getGenero(): string{
        return $this->genero;
    }

    public function getTelefono(): string{
        return $this->telefono;
    }

    public function getPrograma(): string{
        return $this->programa;
    }

    public function getUniversidad(): string{
        return $this->universidad;
    }


    // Setters
    public function setSemestre(int $semestre): void{
        if ($semestre < 1) {
            throw new \InvalidArgumentException("El semestre debe ser mayor o igual a 1.");
        }

        $this->semestre = $semestre;

    }

    public function setEmail(string $email): void{
        if (!filter_var($email, FILTER_VALIDATE_EMAIL)){
            throw new \InvalidArgumentException("El correo electronico no es válido.");
        }
        
        $this->email = $email;

    }

    public function setTelefono(string $telefono): void{
        if (strlen($telefono) < 7){
            throw new \InvalidArgumentException("El teléfono debe tener al menos 7 digitos.");
        }

        $this->telefono = $telefono;

    }

}

File: backend-estudiantes/src/main/java/com/jcaa/hexagonal/domain/entities/Estudiante.java
package com.jcaa.hexagonal.domain.entities;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "estudiante")
public class Estudiante {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String nombre;
    private String apellido;
    private LocalDate fechaNacimiento;
    private Integer semestre;
    private String email;
    private String genero;
    private String telefono;
    private String programa;
    private String universidad;

    // Constructor vacío
    public Estudiante() {}

    // Constructor con argumentos
    public Estudiante(Integer id, String nombre, String apellido, LocalDate fechaNacimiento,
                      Integer semestre, String email, String genero,
                      String telefono, String programa, String universidad) {
        this.id = id;
        this.nombre = nombre;
        this.apellido = apellido;
        this.fechaNacimiento = fechaNacimiento;
        this.semestre = semestre;
        this.email = email;
        this.genero = genero;
        this.telefono = telefono;
        this.programa = programa;
        this.universidad = universidad;
    }


    // ----- Getters -----

    public Integer getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public LocalDate getFechaNacimiento() {
        return fechaNacimiento;
    }

    public Integer getSemestre() {
        return semestre;
    }

    public String getEmail() {
        return email;
    }

    public String getGenero() {
        return genero;
    }

    public String getTelefono() {
        return telefono;
    }

    public String getPrograma() {
        return programa;
    }

    public String getUniversidad() {
        return universidad;
    }


    // ----- Setters. -----

    public void setId(Integer id) {
        this.id = id;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public void setFechaNacimiento(LocalDate fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public void setSemestre(Integer semestre) {
        this.semestre = semestre;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setGenero(String genero) {
        this.genero = genero;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public void setPrograma(String programa) {
        this.programa = programa;
    }

    public void setUniversidad(String universidad) {
        this.universidad = universidad;
    }}
